### PR TITLE
Feature/XIVY-11623 Use new core API to async start processes and start managed reader thread

### PR DESCRIPTION
**Pull request by weissreto**

### CODE REVIEW

File: kafka-connector-test/pom.xml [hunk 1]
```diff
@@ -1,31 +1,42 @@
...
   <artifactId>kafka-connector-test</artifactId>
-  <version>10.0.9-SNAPSHOT</version>
+  <version>11.2.0-SNAPSHOT</version>
   <packaging>iar</packaging>
   <dependencies>
     <dependency>
       <groupId>com.axonivy.connector.kafka</groupId>
       <artifactId>kafka-connector</artifactId>
-      <version>10.0.9-SNAPSHOT</version>
+      <version>${project.version}</version>
```
Comment (ivy-rew) [inline kafka-connector-test/pom.xml:13]: :liberia:

File: kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java [hunk 2]
```diff
@@ -36,15 +35,11 @@
  * {@link IProcessStartEventBean} to listen to Apache Kafka topics.
  */
 public class KafkaStartEventBean extends AbstractProcessStartEventBean {
-	private static final AtomicInteger consumerBeanCounter = new AtomicInteger(0);
 	private static final String KAFKA_CONFIGURATION_NAME_FIELD = "kafkaConfigurationNameField";
 	private static final String TOPIC_PATTERN_FIELD = "topicPatternField";
 	private static final String SYNCHRONOUS_FIELD = "synchronousField";
-	private static ExecutorService workerExecutor;
-	private ExecutorService consumerExecutor;
-	private KafkaConsumerRunnable consumerThread;
```
Comment (ivy-rew) [inline kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java:45]: :1st_place_medal:

**Review by ivy-rew — APPROVED** [review]
thanks

File: kafka-connector-test/pom.xml (hunk 1 above)
Comment (weissreto) [reply kafka-connector-test/pom.xml:13]: What does LR mean?

Comment (ivy-rew) [reply kafka-connector-test/pom.xml:13]: was just posting a victorious flag; cheering your efforts to simplify the dependencies here :)
sorry, will try to use less emojis. Always forget that windows users can't see it

File: kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java [hunk 3]
```diff
@@ -145,123 +106,121 @@ protected boolean isSynchronous() {
...
-				}
+				throw new RuntimeException("Exception while listening for topic.", e);
 			}
 		}
 
-		public void wakeup() {
-			if(consumer != null) {
-				consumer.wakeup();
+		
+		private void startProcess(ConsumerRecord<?, ?> record, KafkaConsumer<?, ?> consumer2, boolean synchronous) {
```
Comment (peters-axon) [inline kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java:163]: consumer2 is obsolete

File: kafka-connector-test/pom.xml (hunk 1 above)
Comment (weissreto) [reply kafka-connector-test/pom.xml:13]: I can see it using firefox but some are missing with Edge :-(

File: kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java (hunk 3 above)
Comment (weissreto) [reply kafka-connector/src/com/axonivy/connector/kafka/KafkaStartEventBean.java]: Good catch. Don't know why the compiler did not warn me about this ...

**Review by peters-axon — APPROVED** [review]

**Pull request merged**